### ISSUE TITLE
Unsafe Connection

### ISSUE THREAD

**Issue by FelixGail** [EdyTheCow/docker-pterodactyl#14]
When using the panel to create/edit nodes the request is sent over an unsafe http connection. Which is only upgraded afterwards by my traefik configuration.

Initial request which is redirected by my traefik redirection scheme and follow up request:
![ptero](https://user-images.githubusercontent.com/20506647/132241602-e7995d8d-2f6c-48d9-b14f-8077bfef6b6d.jpg)

Is there a way to fix this?

**Comment by EdyTheCow** [comment]
Hey, 
What I am suspecting is panel using http only requests because LE_EMAIL option isn't defined https://github.com/EdyTheCow/docker-pterodactyl/blob/master/panel/compose/docker-compose.yml#L17 so it defaults to using http. I think I ran into a similar issue before. To be honest, it doesn't really make much sense to run panel under Traefik anyway. What I did was fill out the LE_EMAIL with an email used to generate Let's Encrypt certificate and simply point cloudflare to panel's IP. That also means you cannot use Traefik because panel it's is handling the certificates, unless you do some configuration to ignore certificate generation from Traefik's end.

So basically panel doesn't use http requests unless LE_EMAIL is defined.

**Comment by fenpaws** [comment]
Well that's not entirely true, i use the panel behind my traefik instance since I host a lot of services. I post my docker-compose.yml here. 

This works flawlessly, well besides the issue #13. But I also described how I fixed that.

```yml
version: '3.8'
x-common:
  database:
    &db-environment
    # Do not remove the "&db-password" from the end of the line below, it is important
    # for Panel functionality.
    MYSQL_PASSWORD: &db-password "XXXXXXXX"
    MYSQL_ROOT_PASSWORD: "XXXXXXXX"
  panel:
    &panel-environment
    APP_URL: "https://panel.XXXXXXXX.XX"
    # A list of valid timezones can be found here: http://php.net/manual/en/timezones.php
    APP_TIMEZONE: "Europe/Berlin"
    APP_SERVICE_AUTHOR: "no-reply@XXXXXXXX.cc"
    # Uncomment the line below and set to a non-empty value if you want to use Let's Encrypt
    # to generate an SSL certificate for the Panel.
    # LE_EMAIL: ""
  mail:
    &mail-environment
    MAIL_FROM: "noreply@example.com"
    MAIL_DRIVER: "smtp"
    MAIL_HOST: "mail"
    MAIL_PORT: "1025"
    MAIL_USERNAME: ""
    MAIL_PASSWORD: ""
    MAIL_ENCRYPTION: "true"

#
# ------------------------------------------------------------------------------------------
# DANGER ZONE BELOW
#
# The remainder of this file likely does not need to be changed. Please only make modifications
# below if you understand what you are doing.
#

services:

  # traefik:
  #   image: traefik:v2.3
  #   restart: always
  #   ports:
  #     - "80:80"
  #     - "443:443"
  #   volumes:
  #     - "/var/run/docker.sock:/var/run/docker.sock"
  #     - "${DATA_DIR}/traefik/traefik.toml:/etc/traefik/traefik.toml"
  #     - "${DATA_DIR}/traefik/acme.json:/acme.json"

  database:
    image: library/mysql:8.0
    restart: always
    command: --default-authentication-plugin=mysql_native_password
    volumes:
      - "${DATA_DIR}/database:/var/lib/mysql"
    environment:
      <<: *db-environment
      MYSQL_DATABASE: "panel"
      MYSQL_USER: "pterodactyl"

  cache:
    image: redis:alpine
    restart: always

  panel:
    image: ghcr.io/pterodactyl/panel:v1.4.2
    restart: always
    links:
      - database
      - cache
    volumes:
      - "${DATA_DIR}/panel/var/:/app/var/"
      - "${DATA_DIR}/panel/nginx/:/etc/nginx/conf.d/"
      - "${DATA_DIR}/panel/certs/:/etc/letsencrypt/"
      - "${DATA_DIR}/panel/logs/:/app/storage/logs"
    labels:
      - "traefik.enable=true"
      # HTTPS
      - traefik.http.routers.pterodactyl-panel_ssl.rule=Host(`${TRAEFIK_PANEL_DOMAIN}`)
      - traefik.http.routers.pterodactyl-panel_ssl.entrypoints=websecure
      - traefik.http.routers.pterodactyl-panel_ssl.tls.certresolver=letsencrypt
      - traefik.http.services.pterodactyl-panel.loadbalancer.server.port=80
      # HTTP
      - traefik.http.routers.pterodactyl-panel.rule=Host(`${TRAEFIK_PANEL_DOMAIN}`)
      - traefik.http.routers.pterodactyl-panel.entrypoints=web
      - traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https
      - traefik.http.routers.pterodactyl-panel.middlewares=redirect-to-https
      
    environment:
      <<: *panel-environment
      <<: *mail-environment
      DB_PASSWORD: *db-password
      APP_ENV: "production"
      APP_ENVIRONMENT_ONLY: "false"
      CACHE_DRIVER: "redis"
      SESSION_DRIVER: "redis"
      QUEUE_DRIVER: "redis"
      REDIS_HOST: "cache"
      DB_HOST: "database"

networks:
  default:
    ipam:
      config:
        - subnet: 172.20.0.0/16

```

**Comment by EdyTheCow** [comment]
Thanks for posting your docker-compose! I'll make sure to test it out later. Do you also run wings under Traefik?

**Comment by fenpaws** [comment]
Yep i do, in the Issue #13 you can find it

**Comment by EdyTheCow** [comment]
Issues reported here were now addressed in the latest commit. The instructions / guide will be updated soon since big parts of the setup structure was changed. The updated setup was tested and is currently running in production environment.
